ml-agents/mlagents/trainers/components/reward_signals/reward_signal.py

```diff
@@ -0,0 +1,68 @@
+import logging
+from mlagents.trainers.trainer import UnityTrainerException
+from mlagents.trainers.policy import Policy
+from collections import namedtuple
+import numpy as np
+import abc
+
+import tensorflow as tf
+
+logger = logging.getLogger("mlagents.trainers")
+
+RewardSignalResult = namedtuple(
+    "RewardSignalResult", ["scaled_reward", "unscaled_reward"]
+)
+
+
+class RewardSignal(abc.ABC):
+    def __init__(self, policy: Policy, strength, gamma):
+        """
+        Initializes a reward signal. At minimum, you must pass in the policy it is being applied to, 
+        the reward strength, and the gamma (discount factor.)
+        :param policy: The Policy object (e.g. PPOPolicy) that this Reward Signal will apply to. 
+        :param strength: The strength of the reward. The reward's raw value will be multiplied by this value. 
+        :param gamma: The time discounting factor used for this reward. 
+        :return: A RewardSignal object. 
+        """
+        class_name = self.__class__.__name__
+        short_name = class_name.replace("RewardSignal", "")
+        self.stat_name = f"Policy/{short_name} Reward"
+        self.value_name = f"Policy/{short_name} Value Estimate"
+        self.gamma = gamma
+        self.policy = policy
+        self.strength = strength
+
+    def evaluate(self, current_info, next_info):
```

Comment:
Can you specify the return type is RewardSignalResult for clarity.
Also, I am wondering if the scaling of the reward signal should be handled by the Reward signal or the trainer.

Reply:
Added the RewardSignalResult into the comment. 

Hmm, we could go either way. 
Pros of the current way: Trainer doesn't have to be aware of the strength, "Strength" could be defined differently for different reward signals.
Pros of doing it in the Trainer: more generic, would be much easier to add normalization of rewards in the future if we want to.